### PR TITLE
Fix adding and moving fields between fieldsets.

### PR DESCRIPTION
**Pull request by frapell**
fixes gh-86 and gh-65

### CODE REVIEW

**Comment by frapell** [comment]
@jenkins-plone-org please run jobs

**Comment by frapell** [comment]
This needs https://github.com/plone/mockup/pull/1203

File: plone/schemaeditor/browser/schema/schema_listing.pt [hunk 1]
```diff
@@ -74,14 +74,16 @@
...
                   tal:attributes="id string:fieldset-${fieldset_name};
-                                  class string:kssattr-fieldset-${fieldset_name}">
+                                  class string:kssattr-fieldset-${fieldset_name};
+                                  data-can-add-fields python:(repeat['group'].start or can_delete) and 'true' or 'false'">
 
           <legend tal:define="group_name python:group.label or view.default_fieldset_label"
                   tal:content="group_name">Fieldset name</legend>
 
-          <a id="delete-fieldset-${fieldset_name}" tal:condition="python:context.enableFieldsets and group.__name__ != view.__name__"
+          <a id="delete-fieldset-${fieldset_name}" tal:condition="python:context.enableFieldsets and can_delete"
```
Comment (petschki) [inline plone/schemaeditor/browser/schema/schema_listing.pt:86]: I would recommend to add the following bootstrap classes here: `btn btn-sm btn-danger`

File: plone/schemaeditor/interfaces.py [hunk 2]
```diff
@@ -157,6 +158,12 @@ def isValidFieldName(value):
 
 class INewField(Interface):
 
+    fieldset_id = Int(
+        title=_(u'Fieldset ID'),
+        description=_(u'Used to decide where to put this field.'),
+        required=True,
+    )
+
```
Comment (petschki) [inline plone/schemaeditor/interfaces.py:166]: How does this behave for already customized schematas? is there a migration step needed which sets the `fieldset_id` ?

Comment (petschki) [reply plone/schemaeditor/interfaces.py:166]: I don't have an example with already customized schematas. Maybe @mauritsvanrees could give this a try?

**Review by petschki — CHANGES_REQUESTED** [review]

File: plone/schemaeditor/interfaces.py (hunk 2 above)
Comment (frapell) [reply plone/schemaeditor/interfaces.py:166]: @petschki My understanding is that the `INewField` interface is only used to get the fields for the "Add Field" form, and then what persists is the field itself (chosen with the `factory` field of this form). The `fieldset_id` is not passed in, so everything should continue to work just fine.

**Comment by frapell** [comment]
@jenkins-plone-org please run jobs

**Comment by mauritsvanrees** [comment]
Jenkins tests fail. Some probably random, but some real ones in `plone.restapi` as well. Sample error when I go in with a pdb:

```
(Pdb) pp response.json()
{'message': "'NoneType' object has no attribute 'get'",
 'traceback': ['File '
               '"/Users/maurits/community/plone-coredev/6.0-37/src/Zope/src/ZPublisher/WSGIPublisher.py", '
               'line 167, in transaction_pubevents',
               '    yield',
               '',
               '  File '
               '"/Users/maurits/community/plone-coredev/6.0-37/src/Zope/src/ZPublisher/WSGIPublisher.py", '
               'line 376, in publish_module',
               '    response = _publish(request, new_mod_info)',
               '',
               '  File '
               '"/Users/maurits/community/plone-coredev/6.0-37/src/Zope/src/ZPublisher/WSGIPublisher.py", '
               'line 279, in publish',
               '    bind=1)',
               '',
               '  File '
               '"/Users/maurits/community/plone-coredev/6.0-37/src/Zope/src/ZPublisher/mapply.py", '
               'line 85, in mapply',
               '    return debug(object, args, context)',
               '',
               '  File '
               '"/Users/maurits/community/plone-coredev/6.0-37/src/Zope/src/ZPublisher/WSGIPublisher.py", '
               'line 68, in call_object',
               '    return obj(*args)',
               '',
               '  File '
               '"/Users/maurits/shared-eggs/cp37m/plone.rest-2.0.0a5-py3.7.egg/plone/rest/service.py", '
               'line 22, in __call__',
               '    return self.render()',
               '',
               '  File '
               '"/Users/maurits/community/plone-coredev/6.0-37/src/plone.restapi/src/plone/restapi/services/__init__.py", '
               'line 19, in render',
               '    content = self.reply()',
               '',
               '  File '
               '"/Users/maurits/community/plone-coredev/6.0-37/src/plone.restapi/src/plone/restapi/services/types/add.py", '
               'line 55, in reply',
               '    res = add_field(context, self.request, data)',
               '',
               '  File '
               '"/Users/maurits/community/plone-coredev/6.0-37/src/plone.restapi/src/plone/restapi/types/utils.py", '
               'line 457, in add_field',
               '    add.form_instance.add(field)',
               '',
               '  File '
               '"/Users/maurits/community/plone-coredev/6.0-37/src/plone.schemaeditor/plone/schemaeditor/browser/schema/add_field.py", '
               'line 74, in add',
               "    fieldset_widget = self.widgets.get('fieldset_id')"],
 'type': 'AttributeError'}
```

Apparently, in `add_field.py` `self.widgets` is None.

**Comment by petschki** [comment]
thanks @mauritsvanrees ... I've added a check for `self.widgets` and my local tests are now green again. Let's see what Jenkins says.

**Review by mauritsvanrees — APPROVED** [review]
This in combination with the plone.staticresource PR works for me when I try it locally. Thanks!

Jenkins 3.8 is green. I have restarted 3.7 and 3.9 again. We should probably merge anyway. Afterwards see if we can somehow make these robot tests more stable.

**Comment by mauritsvanrees** [comment]
3.7 has failed, but both 3.8 and 3.9 are green. I merge

**Pull request merged**